### ISSUE TITLE
cy.get('element').should('not.exist') fails on cy.get.

### ISSUE THREAD

**Issue by kzahozhy** [cypress-io/cypress#23055]
### Current behavior

During the test scenario I open target modal dialog, input some text and click button Save. After that dialog is closed, which I check with following instruction:
cy.get('div .modal-dialog').should('not.exist');

But suprisingly this instruction fales on cy.get('div .modal-dialog') with exception: Timed out retrying after 30000ms: Expected to find element: div .modal-dialog, but never found it.
In my opinion it looks like a bug, cause absense of element is what I'm trying to check and the the check is failed because target element absense.

### Desired behavior

cy.get('element').should('not.exist') should pass without any exception.

### Test code to reproduce

cy.get('element').should('not.exist');

### Cypress Version

 Cypress: 9.7.0

**Comment by AtofStryker** [comment]
Hey @kzahozhy. Thank you for opening an issue. Would you be able to provide a reproduction repository to verify the behavior you are seeing?

**Comment by kzahozhy** [comment]
Hello @AtofStryker. Unfortunately I can't provide the original test case, as it operates with the customer application. But I try to find some other place where this issue reproduced. Until the I can only provide details on how and when indirectly.

**Comment by kzahozhy** [comment]
Hello again. I found the cause of the problem. But still do not know what to do with it:

after clicking button Save the dialog hides.  The **cy.get('dialog').should('not.exist');**  works as following: first finds element and then makes sure that it dissapears from the page, but in my case the dialog hides so quick that on the moment of calling cy.get('dialog') it's already dissapear from the page and so **cy.get('dialog').should('not.exist');** fails.

I'm not sure but probably it would be good if **cy.get('dialog').should('not.exist');** passes not only in case if element dissapears after calling cy.get but also if cy.get() fails due to element not found.  I really can't imagine the case where one would call **cy.get('element').should('not.exist');** without previous work with that element, in other words making sure it exists, so some kind of fake pass is not possible.

Meanwhile may be you could advice me some workaround?

**Comment by kzahozhy** [comment]
@AtofStryker  so the easiest way to reproduce my issue is to call cy.get('element').should('not.exist'); on any element that does not exist on the page at all (some fake element).

**Comment by AtofStryker** [comment]
> 

Have you tried  `cy.get('dialog').should('not.be.visible')`? Assuming there is no race condition and the element does exist, but is hidden, this should work.

**Comment by nagash77** [comment]
Unfortunately we have to close this issue due to inactivity. Please comment if there is new information to provide concerning the original issue and we can reopen.

**Comment by kzahozhy** [comment]
Hello. I think I misled you somewhere during explanation. So I try once again: 
I have following instruction in the test - **cy.get('div .modal-dialog').should('not.exist')**. And it fails on **cy.get('div .modal-dialog')** with exception: Timed out retrying after 30000ms: Expected to find element: div .modal-dialog, but never found it.

How can this instruction **cy.get('div .modal-dialog').should('not.exist')** fail because of element absence if it's goal, essence, is to check that element is trully absent?

**Comment by AtofStryker** [comment]
@kzahozhy that should work, but it is a bit difficult to see what is happening without a reproducible reproduction. It seems like some type of race condition. I believe you mentioned this is a customer code base, but are you able to reproduce this outside of that context in a standalone repository? Is your `defaultCommandTimeout` indeed to set to `30000`ms?

**Comment by kzahozhy** [comment]
No, unfortunately I wasn't able to reproduce it for some other app.  
Indeed my default timeout set to 30000ms.

**Comment by AtofStryker** [comment]
unfortunately without a reproduction, I don't think there is going to be much we can do here. If you are able to provide a reproduction we can reopen this issue, but for now I am going to close it until otherwise.

**Comment by andretauffer** [comment]
Hi @kzahozhy, I've got the same issue when updating cypress from 9.7 to 12. I used the same syntax to evaluate that an element disappeared from the view, cy.get('element').should('not.exist'). On the new version I've got this error as well, that cy.get would fail. My approach to get around this was invoking cy.get("element").as("alias") while the element was still existing. Then I could use the alias later on in cy.get("@alias").should("not.exist"). Now it seems to work. Or at least cy.get doesn't fail anymore...

**Comment by nagash77** [comment]
Hi @andretauffer , if you have a reproducible example you can share with us please do comment back and we will be happy to reopen this issue and investigate.  But as is we have not been able to reproduce this behavior and thus have not been able to fix it.

**Comment by Pedro-Mac** [comment]
Hello 👋

I've looked up many places to see if I could find the reason why I can not do `cy.get(element).as('alias')` if that element is not rendered yet. Is that intentional is there something that I am doing wrong on my side?

So, what I am trying to do is

```
cy.get('[data-cy=yadayada').as('aliasName')

cy.get('@aliasName').should('not.exist');
```

And I get the error `Expected to find element [X], but never found it`

Version 12.14.0

**Comment by nagash77** [comment]
@Pedro-Mac Issues in the repo are reserved for bugs and feature requests.  Check out the Cypress Discord Community [![Discord chat](https://img.shields.io/badge/chat-on%20Discord-brightgreen)](https://on.cypress.io/discord), it can be helpful for debugging or answering questions on how to use Cypress.
